### PR TITLE
Request to combine code refactor and party system

### PR DESCRIPTION
**Pull request by MyChakoTaco**
Refactor to use bot.command instead of checking every message and seeing what the prefix is

Added a party system that can be called by anyone and joined/left by reaction emojis

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,4 @@
 __pycache__/
+.env
+.vscode/launch.json
+.env
```
Comment (danny-tom) [inline .gitignore]: Duplicate .env

File: roles.py [hunk 2]
```diff
@@ -6,12 +6,33 @@
...
+    size=8,
+    emoji='❄️',
+    imageURL='https://pbs.twimg.com/profile_images/1084835909185372160/QzmW7PC8.jpg')
 
 ROLES_LIST = [
-              VALORANT,
-              OVERWATCH,
-              PROJECTWINTER
-              ]
+    VALORANT,
```
Comment (danny-tom) [inline roles.py:40]: Are you using a linter? I've been using Flake8 if we want to standardize these indents and other stuff.

File: party.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            self.size = DEFAULT_PARTY_SIZE if size is None else size
+            self.imageURL = None
+            self.joinEmoji = DEFAULT_JOIN_EMOJI
+
+        self.closeEmoji = DEFAULT_CLOSE_EMOJI
+
+    @staticmethod
+    def __getPreset(name):
+        for preset in roles.ROLES_LIST:
+            if name.lower() == preset.name.lower():
```
Comment (danny-tom) [inline party.py]: casefold for safety? I dunno. https://stackoverflow.com/questions/45745661/python-lower-vs-casefold-in-string-matching-and-converting-to-lowercase

File: charon.py [hunk 4]
```diff
@@ -221,4 +185,44 @@ async def on_member_join(member):
...
+        if party.isMatchJoinEmoji(reaction):
+            party.removeMember(user.name)
+            await reaction.message.edit(embed = party.getEmbed())
+            break
+
+# seconds of running background loop
+BACKGROUND_LOOP_TIME = 60
+# This function checks if parties are inactive
+# and cleans up them up if they are
+async def update_parties():
```
Comment (danny-tom) [inline charon.py:253]: There is a decorator that might be available here
https://discordpy.readthedocs.io/en/latest/ext/tasks/index.html?highlight=task#discord.ext.tasks.loop

**Review by danny-tom — CHANGES_REQUESTED** [review]
Mostly looking to standardize some spacing. Some other nitpicks

File: roles.py (hunk 2 above)
Comment (MyChakoTaco) [reply roles.py:40]: I'm just using microsoft visual code, but i've been eyeballing it. i can probably find the Flake8 thing and use it.

File: .gitignore (hunk 1 above)
Comment (MyChakoTaco) [reply .gitignore]: Will fix

File: party.py (hunk 3 above)
Comment (MyChakoTaco) [reply party.py]: casefold sounds good. i'm not sure if roles can have emoji's, but let's just opt for safer

File: charon.py (hunk 4 above)
Comment (MyChakoTaco) [reply charon.py:253]: will check it out

**Comment by MyChakoTaco** [comment]
Made the requested changes

**Review by danny-tom — APPROVED** [review]
Thanks for making the changes. LGTM

**Pull request merged**